include/cuco/detail/static_map_kernels.cuh

```diff
@@ -83,13 +83,13 @@ __global__ void insert(InputIt first,
   __shared__ typename BlockReduce::TempStorage temp_storage;
   std::size_t thread_num_successes = 0;
   
-  auto tid = blockDim.x * blockIdx.x + threadIdx.x;
+  auto tid = static_cast<uint64_t>(blockDim.x) * blockIdx.x + threadIdx.x;
```

Comment:
Let's make the `block_size` template param be `std::size_t`, then we don't need to do any casting. 
```suggestion
  auto tid = block_size * blockIdx.x + threadIdx.x;
```